src/openms/source/METADATA/AbsoluteQuantitationStandards.cpp

```diff
@@ -28,34 +28,77 @@
 // ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 //
 // --------------------------------------------------------------------------
-// $Maintainer: Douglas McCloskey $
-// $Authors: Douglas McCloskey $
+// $Maintainer: Douglas McCloskey, Pasquale Domenico Colaianni $
+// $Authors: Douglas McCloskey, Pasquale Domenico Colaianni $
 // --------------------------------------------------------------------------
 
 #include <OpenMS/METADATA/AbsoluteQuantitationStandards.h>
 
-#include <cstddef> // for size_t & ptrdiff_t
-#include <vector>
-#include <string>
-
 namespace OpenMS
 {
-    
-    AbsoluteQuantitationStandards::AbsoluteQuantitationStandards()
+  bool AbsoluteQuantitationStandards::findComponentFeature_(
+    const FeatureMap& feature_map,
+    const String& component_name,
+    Feature& feature_found
+  ) const
+  {
+    for (const Feature& feature : feature_map)
     {
+      for (const Feature& subordinate : feature.getSubordinates())
+      {
+        if (subordinate.metaValueExists("native_id") && subordinate.getMetaValue("native_id") == component_name)
+        {
+          feature_found = subordinate;
+          return true;
+        }
+      }
     }
-    
-    AbsoluteQuantitationStandards::~AbsoluteQuantitationStandards()
+    return false;
+  }
+
+  void AbsoluteQuantitationStandards::mapComponentsToConcentrations(
+    const std::vector<AbsoluteQuantitationStandards::runConcentration>& run_concentrations,
+    const std::vector<FeatureMap>& feature_maps,
+    std::map<String, AbsoluteQuantitationStandards::featureConcentration>& components_to_concentrations
+  ) const
+  {
+    components_to_concentrations.clear();
+    for (const AbsoluteQuantitationStandards::runConcentration& run : run_concentrations)
     {
+      if (run.sample_name == "" || run.component_name == "")
+      {
+        continue;
+      }
+      for (const FeatureMap& fmap : feature_maps) // not all elements are necessarily processed (break; is present inside the loop)
+      {
+        AbsoluteQuantitationStandards::featureConcentration fc;
+        if (!fmap.metaValueExists("sample_name") || fmap.getMetaValue("sample_name") != run.sample_name) // if the FeatureMap doesn't have a sample_name, or if it is not the one we're looking for: skip.
```

Comment:
@timosachsenberg and @hroest we are currently using a new `metaValue` in `Feature` called "sample_name" to map to different `FeatureMaps`.  Is there a standardized `Feature` attribute that can be used for this?  Alternatively, is there another convention that you or others have used to map to different `FeatureMaps`?